### PR TITLE
[IDP-2068] Automerge official dependency from microsoft

### PR DESCRIPTION
**Pull request by PrincessMadMath**
Add automerge rule for official Microsoft dependency.

**Notes**

We need to use the [Branch ](https://docs.renovatebot.com/key-concepts/automerge/#branch-vs-pr-automerging) approach to avoid the Pull Request approval limitation.  This means we need to allow the renovate bot to push on main without approval. We should still have a branch policy that every push on main must have a successful CI.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -18,4 +18,4 @@ jobs:
       - name: Test with the dotnet CLI
         run: dotnet test tests 
         env:
-          TEST_GITHUB_TOKEN: ${{secrets.TEST_GITHUB_TOKEN}}
\ No newline at end of file
+          GH_TOKEN: ${{secrets.TEST_GITHUB_MG}}
```
Comment (meziantou) [inline .github/workflows/ci.yml:21]: To be updated once IT creates the new token (could be in another PR)

File: README.md [hunk 2]
```diff
@@ -17,7 +17,17 @@ The shared configurations are baselines. Each project is free to set their own r
...
-# Azure DevOps
+#### Enabling Auto-Merge Functionality for GitHub
+
+Auto-merge has been set up using the [branch approach](https://docs.renovatebot.com/key-concepts/automerge/#branch-vs-pr-automerging), chosen to minimize noise and allow for the bypassing of PR review requirements.
+
+For those utilizing branch protection rules on the default branch, specific adjustments are necessary to facilitate auto-merge capabilities for GitHub:
+
+1. **Update Branch Policies**: Modify your branch protection settings to permit the account or service running Renovate to bypass pull request review requirements.
+
+2. **Handling Status Checks**: If your repository enforces "Require status checks to pass before merging," be aware that Renovate will be unable to merge changes into the main branch if any of these checks fail. It will fallback to create a PR.
```
Comment (meziantou) [inline README.md:28]:
```suggestion
2. **Handling Status Checks**: If your repository enforces "Require status checks to pass before merging", be aware that Renovate will be unable to merge changes into the target branch if any of these checks fail. It will fallback to create a pull request.
```

File: tests/TestContext.cs [hunk 3]
```diff
@@ -76,20 +146,31 @@ public async Task AssertPullRequests(string? expected = null, [CallerFilePath] s
...
+        var commits = await GetCommits();
+        InlineSnapshot
+            .WithSettings(settings => settings.ScrubLinesWithReplace(line => Regex.Replace(line, "to [^ ]+ ?", " to redacted")))
+            // ReSharper disable ExplicitCallerInfoArgument
+            .Validate(commits, expected, filePath, lineNumber);
+        // ReSharper restore ExplicitCallerInfoArgument
+    }
 
-    public async Task<IEnumerable<PullRequestInfos>> GetPullRequests()
+    public async Task<IEnumerable<PullRequestInfos>>  GetPullRequests()
```
Comment (meziantou) [inline tests/TestContext.cs:161]:
```suggestion
    public async Task<IEnumerable<PullRequestInfos>> GetPullRequests()
```

File: tests/TestContext.cs [hunk 4]
```diff
@@ -76,20 +146,31 @@ public async Task AssertPullRequests(string? expected = null, [CallerFilePath] s
...
         var pullRequestsInfos = new List<PullRequestInfos>(pullRequests.Count);
 
-        foreach (var pullRequest in pullRequests.OrderBy(x => x.Title))
+        foreach (PullRequestSimple pullRequest in pullRequests.OrderBy(x => x.Title))
         {
-            MarkdownDocument markdownDocument = Markdown.Parse(pullRequest.Body, pipeline);
+            MarkdownDocument markdownDocument = Markdown.Parse(pullRequest.Body!, pipeline);
             var prTitle = pullRequest.Title;
-            var prLabels = pullRequest.Labels.Select(x => x.Name).Order();
+            var prLabels = pullRequest.Labels!.Select(x => x.Name).Order();
```
Comment (meziantou) [inline tests/TestContext.cs:173]: I think you can allow null values everywhere. This is better than using `!` as we don't really care about null here. The only use-case of this is the snapshot generation.

```suggestion
            var prLabels = pullRequest.Labels?.Select(x => x.Name).Order();
```

File: tests/TestContext.cs [hunk 5]
```diff
@@ -105,24 +186,80 @@ public async Task<IEnumerable<PullRequestInfos>> GetPullRequests()
...
+        return commitInfos;
+    }
+    
+    public async Task WaitForLatestCommitChecksToSucceed()
+    {
+        var branches = await gitHubClient.Repos[RepositoryOwner][RepositoryName].Branches.GetAsync() ?? [];
+
+        foreach (var branch in branches)
+        {
+            if (!string.Equals(branch.Name, DefaultBranchName))
```
Comment (meziantou) [inline tests/TestContext.cs:222]:
```suggestion
            if (branch.Name != DefaultBranchName)
```

File: tests/TestContext.cs [hunk 6]
```diff
@@ -132,11 +269,10 @@ private async Task CleanupRepository()
 
     private static async Task<string> GetGitHubToken(ITestOutputHelper outputHelper)
     {
-        var token = Environment.GetEnvironmentVariable("TEST_GITHUB_TOKEN");
-
+        var token = Environment.GetEnvironmentVariable("GH_TOKEN");
+
```
Comment (meziantou) [inline tests/TestContext.cs:273]:
```suggestion

```

File: .github/workflows/ci.yml (hunk 1 above)
Comment (PrincessMadMath) [reply .github/workflows/ci.yml:21]: Yes once I have the token I will come update this here

**Review by meziantou — DISMISSED** [review]

**Review by meziantou — APPROVED** [review]
